Userland/Libraries/LibSQL/SQLClient.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2021, Jan de Visser <jan@de-visser.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibIPC/ServerConnection.h>
+#include <SQLServer/SQLClientEndpoint.h>
+#include <SQLServer/SQLServerEndpoint.h>
+
+namespace SQL {
+
+class SQLClient
+    : public IPC::ServerConnection<SQLClientEndpoint, SQLServerEndpoint>
+    , public SQLClientEndpoint {
+    C_OBJECT(SQLClient);
+    virtual ~SQLClient();
+
+    Function<void(int)> on_connected;
+    Function<void(int)> on_disconnected;
+    Function<void(int, int, String const&)> on_connection_error;
+    Function<void(int, int, String const&)> on_execution_error;
+    Function<void(int, int, int, int)> on_executed_update;
```

Comment:
`executed_update` / `on_executed_update` sounds a bit awkward to read, maybe `on_execution_success` or something?
What's the semantic difference between `on_executed_update` and `on_next_result`? Is the latter something that is invoked for e.g. each row returned from a `SELECT`, and the former invoked once the entire statement has completed?

Reply:
`on_executed_update` is called when the statement resulted in the database being updated, due to INSERT/DELETE/UPDATE actions. The message includes the number(s) of rows affected. And `on_next_result ` is called for every row in the result set of a SELECT or INSERT/UPDATE ... RETURNING statement.

I took a lot of cues in the design of this interface from Java's JDBC. It's old by now, but has proven a pretty good abstraction of the way SQL statements are handled in typical applications. JDBC has different methods to execute a statement- executeUpdate() and execute() if memory serves. The only "problem" is that if you have an 'update' type statement that also returns a result set there are some hoops to jump through, hoops which I think I avoided in my design.

Reply:
But I like `execution_success`. I wasn't a big fan of `executed_update` myself but couldn't think of anything better. 

We should probably send an `execution_success` for a `SELECT` as well. That way the app can determine that the statement worked without having to going through maybe some expensive setup to process the result.